way_to_home/custom_user/views.py

```diff
@@ -116,3 +125,64 @@ def signin_google(request):
         return HttpResponse("User was successfully created", status=201)
 
     return HttpResponse("User's data is empty", status=400)
+
+
+@require_http_methods(["POST"])
+def reset_password(request):
+    """Function that provides reset user password"""
+    data = request.body
+    email = data.get('email')
+    if not email_validator(email):
+        return RESPONSE_400_INVALID_EMAIL
+    user = CustomUser.get_by_email(email=email)
+    if not user:
+        return RESPONSE_400_OBJECT_NOT_FOUND
+    token = create_token(data={'email': user.email})
+    send_email_password_update(user, token)
+    return RESPONSE_200_OK
+
+
+@require_http_methods(['PUT'])
+def confirm_reset_password(request, token):
+    """Function that provides confirm reset user password"""
+    data = request.body
+    new_password = data.get('new_password')
+    confirm = decode_token(token)
+
+    if not confirm:
+        return RESPONSE_498_INVALID_TOKEN
+    user = CustomUser.get_by_email(email=confirm.get('email'))
+
+    if not user:
+        return RESPONSE_400_OBJECT_NOT_FOUND
+
+    if not password_validator(new_password):
+        return RESPONSE_400_INVALID_DATA
+
+    if not user.check_password(new_password):
+        is_updated = user.update(password=new_password)
+
+        if not is_updated:
+            return RESPONSE_400_DB_OPERATION_FAILED
+
+        send_successful_update_email(user)
+        return RESPONSE_200_UPDATED
+
+    return RESPONSE_400_INVALID_DATA
+
+
+@require_http_methods(["PUT"])
+def change_password(request):
+    """Function that provides change user password"""
+    user = request.user
+    data = request.body
+    new_password = data.get('new_password')
+    old_password = data.get('old_password')
+    if not new_password or not old_password:
+        return RESPONSE_400_INVALID_DATA
```

Comment:
```suggestion
    if not new_password or not old_password:
        return RESPONSE_400_INVALID_DATA
```
This case will be handled by next 2 lines, so it is not necessary, you may remove it

Reply:
done